File: P1S3/FiltroEncriptacion.cpp
//
// Created by ines on 8/4/21.
//

#include "FiltroEncriptacion.h"

FiltroEncriptacion::FiltroEncriptacion() {}

void FiltroEncriptacion::procesar(Formulario &formulario) {
    unsigned edad = formulario.getEdad();
    string descripcion = formulario.getDescripcion();
    string dni = formulario.getDNI();

    edad = edad*2;

    for(unsigned i = 0; i < descripcion.length();++i)
        descripcion[i] += 2;

    for(unsigned i = 0; i < dni.length();++i)
        dni[i] += 2;

    formulario.setDescripcion(descripcion);
    formulario.setDNI(dni);
    formulario.setEdad(edad);
}

File: P1/P1S3_JAVA/src/FiltroEncriptacion.java
public class FiltroEncriptacion implements Filtro{

    public void procesar(Formulario formulario){
        String desc = formulario.getDescripcion();
        String dni = formulario.getDni();
        int edad = formulario.getEdad();

        for (int i = 0; i < desc.length(); ++i){
            char aux = (char)(desc.charAt(i) + 2);
            desc = desc.substring(0, i) + aux + desc.substring(i+1);
        }

        for (int i = 0; i < dni.length(); ++i){
            char aux = (char)(dni.charAt(i) + 2);
            dni = dni.substring(0, i) + aux + dni.substring(i+1);
        }

        formulario.setDescripcion(desc);
        formulario.setDni(dni);
        formulario.setEdad(edad);
        formulario.setEncriptado(true);
    }
}
